crates/core/src/error.rs

```diff
@@ -1,810 +1,451 @@
+    Password,
+    /// Repository Error
+    Repository,
+    /// Unsupported Feature Error
+    Unsupported,
+    /// Verification Error
+    Verification,
+    /// Virtual File System Error
+    Vfs,
 }
 
-/// [`CommandErrorKind`] describes the errors that can happen while executing a high-level command
-#[derive(Error, Debug, Display)]
-pub enum CommandErrorKind {
-    /// path is no dir: `{0}`
-    PathIsNoDir(String),
-    /// used blobs are missing: blob `{0}` doesn't existing
-    BlobsMissing(BlobId),
-    /// used pack `{0}`: size does not match! Expected size: `{1}`, real size: `{2}`
-    PackSizeNotMatching(PackId, u32, u32),
-    /// used pack `{0}` does not exist!
-    PackNotExisting(PackId),
-    /// pack `{0}` got no decision what to do
-    NoDecision(PackId),
-    /// `{0}`
-    #[error(transparent)]
-    FromParseFloatError(#[from] ParseFloatError),
-    /// `{0}`
-    #[error(transparent)]
-    FromParseIntError(#[from] ParseIntError),
-    /// Bytesize parser failed: `{0}`
-    FromByteSizeParser(String),
-    /// --repack-uncompressed makes no sense for v1 repo!
-    RepackUncompressedRepoV1,
-    /// datetime out of range: `{0}`
-    FromOutOfRangeError(#[from] OutOfRangeError),
-    /// node type `{0:?}` not supported by dump
-    DumpNotSupported(NodeType),
-    /// `{0}`
-    #[error(transparent)]
-    FromJsonError(#[from] serde_json::Error),
-    /// version `{0}` is not supported. Allowed values: {1:?}
-    VersionNotSupported(u32, RangeInclusive<u32>),
-    /// cannot downgrade version from `{0}` to `{1}`
-    CannotDowngrade(u32, u32),
-    /// compression level `{0}` is not supported for repo v1
-    NoCompressionV1Repo(i32),
-    /// compression level `{0}` is not supported. Allowed values: `{1:?}`
-    CompressionLevelNotSupported(i32, RangeInclusive<i32>),
-    /// Size is too large: `{0}`
-    SizeTooLarge(bytesize::ByteSize),
-    /// min_packsize_tolerate_percent must be <= 100
-    MinPackSizeTolerateWrong,
-    /// max_packsize_tolerate_percent must be >= 100 or 0"
-    MaxPackSizeTolerateWrong,
-    /// error creating `{0:?}`: `{1:?}`
-    ErrorCreating(PathBuf, Box<RusticError>),
-    /// error collecting information for `{0:?}`: `{1:?}`
-    ErrorCollecting(PathBuf, Box<RusticError>),
-    /// error setting length for `{0:?}`: `{1:?}`
-    ErrorSettingLength(PathBuf, Box<RusticError>),
-    /// `{0}`
-    #[error(transparent)]
-    FromRayonError(#[from] rayon::ThreadPoolBuildError),
-    /// Conversion from integer failed: `{0:?}`
-    ConversionFromIntFailed(TryFromIntError),
-    /// Not allowed on an append-only repository: `{0}`
-    NotAllowedWithAppendOnly(String),
-    /// Specify one of the keep-* options for forget! Please use keep-none to keep no snapshot.
-    NoKeepOption,
-    /// `{0}`
-    #[error(transparent)]
-    FromParseError(#[from] shell_words::ParseError),
-}
+#[derive(thiserror::Error, Debug)]
+#[non_exhaustive]
+/// Errors that can result from rustic.
+pub struct RusticError {
```

Comment:
What do you think about ordering the `fields` in the same order as we use them in the `Display` implementation?

Reply:
To me this could make it "easier" to make sure all the fields are used in the `impl Disaply`